crates/compiler/can/src/expr.rs

```diff
@@ -572,9 +572,37 @@ pub struct WhenBranchPattern {
     pub degenerate: bool,
 }
 
+#[derive(Clone, Debug, Default)]
+pub struct Refinements(VecMap<Symbol, (Variable, Variable)>);
```

Comment:
What do you think about using a `SmallVec` here instead (there is one in roc_collections)? The hypothesis is that typically, we have 1 or 0 symbols to refine.

Reply:
Sounds like a good idea!